isis/src/base/apps/topds4/topds4.cpp

```diff
@@ -31,9 +33,17 @@ namespace Isis {
     Process p;
     p.SetInputCube(icube);
 
-    // NEED TO WRITE AND CLOSE THE OUTPUT FILE BEFORE RENDERING SO THE FILE SIZE CALLBACK CAN GET THE FINAL FILE SIZE
     QString outputFile = ui.GetFileName("TO");
 
+    // NEED TO WRITE AND CLOSE THE OUTPUT FILE BEFORE RENDERING SO THE FILE SIZE CALLBACK CAN GET THE FINAL FILE SIZE
+
+    // Need a better solution for this.
+    QString outputCubePath = outputFile+".cub";
```

Comment:
How to handle one output filepath indicated by "TO" vs. two output files: the label and the cube itself. `tgocassisrdrgen` uses the one input for both and appends ".xml" or ".img" as needed. Thoughts for this application?

Reply:
I would have to specify the xml label and then strip that off to get the cube name. Ideally it would check if the extension is .xml, .cub, or neither. Then it will strip and add extensions as needed.

Reply:
Most if not all of the apps that deal with .img .xml .lbl, ... ask for the label file then change the extension to get at the other files.